src/main/java/homework_2/random_chars_table/RandomCharsTable.java

```diff
@@ -0,0 +1,55 @@
+package homework_2.random_chars_table;
+
+import java.util.Random;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class RandomCharsTable {
+    public void run() {
+        Scanner scanner = new Scanner(System.in);
+        String[] params = new String[3];
+        for (int i = 0; i < params.length; i++) {
+            params[i] = scanner.next();
+        }
+        if (!isNumber(params[0]) || !isNumber(params[1])) {
+            System.out.println("ERROR! This is not a number");
+            return;
+        }
+        int rows = Integer.parseInt(params[0]);
+        int columns = Integer.parseInt(params[1]);
+        if (rows == 0 || columns == 0) {
+            return;
+        }
+        if (rows < 0 || columns < 0) {
+            System.out.println("ERROR! One of the numbers is negative");
+            return;
+        }
+        String strategy = params[2];
+        int strategyMod;
+        if (strategy.equals("even")) {
```

Comment:
move to a new separate method (getResult, for example)

Reply:
Why

Reply:
Because your code is too complex and difficult to read